packages/dashboards/src/lib/components/timeseries-widget/chart-presets/xy-chart/xy-chart.component.ts

```diff
@@ -70,6 +70,11 @@ export abstract class XYChartComponent extends TimeseriesChartComponent
         return this.configuration.legendPlacement === LegendPlacement.Right;
     }
 
+    /** Checks if the series is interactive or if the dataSource has a link property */
+    public isSeriesInteractive(legendSeries: IChartAssistSeries<IAccessors>): boolean {
+        return !(legendSeries?.link === undefined) || !(legendSeries?.secondaryLink === undefined) || !(this.seriesInteractive === undefined);
```

Comment:
`this.seriesInteractive` is a boolean, so we should use the value as is instead of checking for undefined.